src/sql/workbench/parts/notebook/outputs/notebookMarkdown.ts

```diff
@@ -0,0 +1,240 @@
+		renderer.image = (href: string, title: string, text: string) => {
+			href = this.cleanUrl(!markdown.isTrusted, notebookFolder, href);
+			let dimensions: string[] = [];
+			if (href) {
+				const splitted = href.split('|').map(s => s.trim());
+				href = splitted[0];
+				const parameters = splitted[1];
+				if (parameters) {
+					const heightFromParams = /height=(\d+)/.exec(parameters);
+					const widthFromParams = /width=(\d+)/.exec(parameters);
+					const height = heightFromParams ? heightFromParams[1] : '';
+					const width = widthFromParams ? widthFromParams[1] : '';
+					const widthIsFinite = isFinite(parseInt(width));
+					const heightIsFinite = isFinite(parseInt(height));
+					if (widthIsFinite) {
+						dimensions.push(`width="${width}"`);
+					}
+					if (heightIsFinite) {
+						dimensions.push(`height="${height}"`);
+					}
+				}
+			}
+			let attributes: string[] = [];
+			if (href) {
+				attributes.push(`src="${href}"`);
+			}
+			if (text) {
+				attributes.push(`alt="${text}"`);
+			}
+			if (title) {
+				attributes.push(`title="${title}"`);
+			}
+			if (dimensions.length) {
+				attributes = attributes.concat(dimensions);
+			}
+			return '<img ' + attributes.join(' ') + '>';
+		};
+		renderer.link = (href, title, text): string => {
+			href = this.cleanUrl(!markdown.isTrusted, notebookFolder, href);
+			if (href === null) {
+				return text;
+			}
+			// Remove markdown escapes. Workaround for https://github.com/chjj/marked/issues/829
+			if (href === text) { // raw link case
+				text = removeMarkdownEscapes(text);
+			}
+			title = removeMarkdownEscapes(title);
+			href = removeMarkdownEscapes(href);
+			if (
+				!href
+				|| !markdown.isTrusted
+				|| href.match(/^data:|javascript:/i)
+				|| href.match(/^command:(\/\/\/)?_workbench\.downloadResource/i)
+			) {
+				// drop the link
+				return text;
+
+			} else {
+				// HTML Encode href
+				href = href.replace(/&/g, '&amp;')
+					.replace(/</g, '&lt;')
+					.replace(/>/g, '&gt;')
+					.replace(/"/g, '&quot;')
+					.replace(/'/g, '&#39;');
+				return `<a href=${href} data-href="${href}" title="${title || href}">${text}</a>`;
+			}
+		};
+		renderer.paragraph = (text): string => {
+			return `<p>${text}</p>`;
+		};
+
+		if (options.codeBlockRenderer) {
+			renderer.code = (code, lang) => {
+				const value = options.codeBlockRenderer!(lang, code);
+				// when code-block rendering is async we return sync
+				// but update the node with the real result later.
+				const id = defaultGenerator.nextId();
+
+				const promise = value.then(strValue => {
+					withInnerHTML.then(e => {
+						const span = element.querySelector(`div[data-code="${id}"]`);
+						if (span) {
+							span.innerHTML = strValue;
+						}
+					}).catch(err => {
+						// ignore
+					});
+				});
+
+				if (options.codeBlockRenderCallback) {
+					promise.then(options.codeBlockRenderCallback);
+				}
+
+				return `<div class="code" data-code="${id}">${escape(code)}</div>`;
```

Comment:
Do we need to update CSS for this or is it already correct looking? Can we verify this, just in case it's different from existing markdown. Thanks!

Reply:
I have a number of test notebooks and couldn't see any weirdness from this.